### ISSUE TITLE
Code copy button overlaps code at smaller screen sizes

### ISSUE THREAD

**Issue by talha131** [Pelican-Elegant/elegant#574]
I am not sure how to fix this issue. Suggestions are welcome.

![](https://i.imgur.com/SfiSa3S.png)

Related to #525

**Comment by talha131** [comment]
I guess user can just change the orientation of his screen.

![](https://i.imgur.com/l6tpBy7.png)

**Comment by iranzo** [comment]
@talha131 have a look at https://www.dannyguo.com/blog/how-to-add-copy-to-clipboard-buttons-to-code-blocks-in-hugo/ they put the copy button 'outside' the code block

**Comment by talha131** [comment]
Yeah, I don't think it will fit the UI we have for include code plugin.

https://elegant.oncrashreboot.com/code-snippets-include-code-from-file

**Comment by neirbowj** [comment]
This also appears to affect typical and large screen sizes where the first line in the code block is long. I think I will prepare a separate issue to request a feature to disable the copy button per site, per page, and/or per code-block.

**Comment by talha131** [comment]
@neirbowj, I agree. UI needs improvement. Perhaps, we will have to use @iranzo suggestion.

I decided to release the current UI for a couple of reasons,

1. Most languages have formating tools that break down long lines into smaller chunks. I expect most users will not have long lines if they format their code using such tools
2. I had to release this feature to get user feedback

Now that you have pointed out issues, I will try to fix it in future releases. I am open to suggestions.

For now, an easy (and temporary) workaround is to have a short comment line at the beginning of the code snippet or break the line.

**Comment by silverhook** [comment]
I agree with @iranzo that it might make sense to put it outside of the box.

How about hanging it off the top-right edge? See attached mock-ups:

![elegant_code_snippets_copy_icon_tag](https://user-images.githubusercontent.com/3227264/147418286-d3086683-1bf9-470a-bf5a-a6ede0129efe.png)

![elegant_code_snippets_copy_word_tag](https://user-images.githubusercontent.com/3227264/147418287-613b804c-a34b-43a9-a39b-26466a596f1e.png)

**Comment by iranzo** [comment]
I like the 2nd one :-)

**Comment by silverhook** [comment]
That should be doable with just CSS, no?
